### PR TITLE
feat: optionally return all raw email addresses for user

### PR DESCRIPTION
**Pull request by jaredscheib**
Adopted in part from https://github.com/jaredhanson/passport-github/pull/27

### CODE REVIEW

File: lib/strategy.js [hunk 1]
```diff
@@ -129,10 +131,22 @@ Strategy.prototype.userProfile = function(accessToken, done) {
...
-        if (json[index].primary) {
-          profile.emails = [{ value: json[index].email }];
-          break;
+      if (!json || !json.length) {
+        return done(new Error('Failed to fetch user emails'));
+      }
+
+      if (self._allRawEmails) {
+        profile.emails = json.map(function (email) {
+          var mappedEmail = Object.assign(email, { value: email.email });
```
Comment (ultrabluewolf) [inline lib/strategy.js:140]: How do you know which email is the primary one? Should we add type to these? See: http://www.passportjs.org/docs/profile/

Comment (jaredscheib) [reply lib/strategy.js:140]: here's a sample `json` object:
```
[ { primary: true,
    verified: true,
    visibility: 'private',
    value: 'bob.pineapple@gmail.com' },
  { primary: false,
    verified: true,
    visibility: null,
    value: 'pineapple@users.noreply.github.com' },
  { primary: false,
    verified: true,
    visibility: null,
    value: 'bob@pineapple.com' } ]
```

Comment (ultrabluewolf) [reply lib/strategy.js:140]: Ah, is this what the response looks like? ```[
  {
    "email": "octocat@github.com",
    "verified": true,
    "primary": true,
    "visibility": "public"
  }
]```

Comment (ultrabluewolf) [reply lib/strategy.js:140]: Cool, sounds good.

Comment (jaredscheib) [reply lib/strategy.js:140]: yep, exactly

Comment (jaredscheib) [reply lib/strategy.js:140]: my `map` just transforms these objects to have a `value` key, which seems to be a convention across passport addons

Comment (ultrabluewolf) [reply lib/strategy.js:140]: Should we add a test then?

**Review by ultrabluewolf — APPROVED** [review]
LGTM, but it would be great to add a test as well.

File: lib/strategy.js (hunk 1 above)
Comment (jaredscheib) [reply lib/strategy.js:140]: added tests!

**Review by ultrabluewolf — APPROVED** [review]
:shipit:

**Pull request merged**